src/redux/books/booksSlice.js

```diff
@@ -1,47 +1,58 @@
-const ADD_BOOK = 'Bookstore/books/ADD_BOOK';
-const REMOVE_BOOK = 'Bookstore/books/REMOVE_BOOK';
-
-const initialState = [
-  {
-    id: '1',
-    title: 'The Great Gatsby',
-    author: 'John Smith',
-  },
-  {
-    id: '2',
-    title: 'Anna Karenina',
-    author: 'Leo Tolstoy',
-  },
-  {
-    id: '3',
-    title: 'The Selfish Gene',
-    author: 'Richard Dawkins',
-  },
-];
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import BookSrvc from '../booksAPI';
 
+const initialState = {
+  books: [],
+};
 // Actions
 
-export const addBook = (book) => ({
-  type: ADD_BOOK,
-  payload: book,
-});
+const fetchBooks = createAsyncThunk(
+  'books/fetchBooks',
+  async () => {
+    const { data } = await BookSrvc.getSrvc();
+    return Object.keys(data).map((key) => ({ ...data[key][0], item_id: key }));
+  },
+);
+
+const addBook = createAsyncThunk(
+  'books/addBook',
+  async (Book) => {
+    const response = await BookSrvc.addSrvc(Book);
+    return response.data;
+  },
+);
 
-export const removeBook = (id) => ({
-  type: REMOVE_BOOK,
-  payload: id,
-});
+const deleteBook = createAsyncThunk(
+  'books/deleteBook',
+  async (id) => {
+    await BookSrvc.deleteSrvc(id);
+    return id;
+  },
+);
 
 // Reducer
 
-const bookReducer = (state = initialState, action) => {
+const booksSlice = (state = initialState, action) => {
   switch (action.type) {
-    case ADD_BOOK:
-      return [...state, action.payload];
-    case REMOVE_BOOK:
-      return state.filter((book) => book.id !== action.payload);
+    case 'books/fetchBooks/fulfilled':
+      return {
+        ...state,
+        books: action.payload,
+      };
+    case addBook.fulfilled:
+      return {
+        ...state,
+        books: [...state.books, action.payload],
+      };
+    case deleteBook.fulfilled:
+      return {
+        ...state,
+        books: state.books.filter((book) => book.id !== action.payload),
+      };
     default:
       return state;
   }
 };
```

Comment:
- [ ] To simplify the process of adding data to redux in a standard way, it is recommended to use the redux toolkit's `createSlice` to construct your reducer. Please refer to the article [Understanding createSlice in Redux Toolkit - ReactJS](https://medium.com/geekculture/understanding-createslice-in-redux-toolkit-reactjs-eca8d20f45d7) for guidance on how to use this tool.

Reply:
This is not the project requirement and I will do it after I understand how to implement it